### ISSUE TITLE
assert error in the context_t

### ISSUE THREAD

**Issue by pengweichu** [zeromq/cppzmq#74]
I think in the content_t class, we must check the ptr has been destroyed or not, otherwise if we called the close before destroy, then the ZMQ_ASSERT will be raise when the class destroying.

Below is the modified code:

```
inline ~context_t () ZMQ_NOTHROW
{
    if (!ptr)
    {
        return;
    }

    int rc = zmq_ctx_destroy (ptr);
    ptr = NULL;
    ZMQ_ASSERT (rc == 0);
}

inline void close() ZMQ_NOTHROW
{
    if (!ptr)
    {
        return;
    }

    int rc = zmq_ctx_destroy (ptr);
    ptr = NULL;
    ZMQ_ASSERT (rc == 0);
}
```

**Comment by ignitusboyone** [comment]
According to the man page

http://api.zeromq.org/3-2:zmq-ctx-destroy
http://api.zeromq.org/4-2:zmq-ctx-destroy

The zmq_ctx_destroy() function shall return zero if successful. Otherwise it shall return -1 and set errno to one of the values defined below

So why are we asserting that close and ~context_t were successful.  Shouldn't the assert check should be  rc != 0 or rc == -1 ?

**Comment by sigiesec** [comment]
This issue was fixed by #85
